Code/Framework/AzQtComponents/AzQtComponents/Components/DockBarButton.cpp

```diff
@@ -89,6 +96,15 @@ namespace AzQtComponents
         setFocusPolicy(Qt::ClickFocus);
     }
 
+    bool DockBarButton::eventFilter(QObject* object, QEvent* event)
+    {
+        if (event->type() == QEvent::WindowStateChange)
+        {
+            SetMaximizeRestoreButton();
+        }
```

Comment:
This logic should only run if `m_buttonType == DockBarButton::MaximizeButton`. So could either do that check here, or instead could only install the event filter for that case.